docs/gatewify-an-app.md

```diff
@@ -0,0 +1,169 @@
+          allowed-roles: ADMINISTRATOR
+        - intercept-url: /newapp/backend/public*
+          anonymous: true
+      newappfrontend: 
+        target: http://localhost:80/newapp/frontend
+        access-rules:
+        - intercept-url: /newapp/**
+          anonymous: true
+        headers:
+          proxy: true
+          username: false
+          roles: false
+          org: false
+          orgname: true
+          json-user: true
+```
+
+### Application configuration
+
+#### Headers
+
+As you can see above, the `newappfrontend` service has a header section, which is used to override the default headers that will be sent to the frontend application.
+
+In `gateway.yaml` as the start of the file, you can modify the default headers sent to apps :
+    
+```yaml
+    georchestra:
+      gateway:
+        default-headers:
+          proxy: true
+          ...
+          json-user: true
+```
+
+
+It is also possible to create a specific role in the console app which grants access to the backend, eg with role `NEWAPP_ADMIN`.
+
+#### How it works
+
+With every request, the proxy adds specific HTTP headers, allowing the application to know:
+
+* if the request comes from a registered user, or an anonymous one - this is `sec-username` (not provided if anonymous).
+* which roles the user holds - `sec-roles` is a semi-colon separated list of roles (not provided if anonymous).
+* which organisation the user belongs to - `sec-orgname` provides the human-readable organisation title while `sec-org` is mapped onto the organisation id (LDAP's `cn`).
+
+Several other user properties are also provided as headers:
+
+* `sec-proxy` tells that request comes from the proxy
+* `sec-email` is the user email
+* `sec-firstname` is the first name (LDAP `givenName`)
+* `sec-lastname` is the second name (LDAP `sn`)
+* `sec-tel` is the user phone number (LDAP `telephoneNumber`)
+
+You can find full configuration in [HeaderMappings.java](https://github.com/georchestra/georchestra-gateway/blob/main/gateway/src/main/java/org/georchestra/gateway/model/HeaderMappings.java) file. You just need to rename fields from camelCase (in java file) to kebab-case (in yaml file).
+
+The application handles requests appropriately thanks to the headers received.
+Some applications will require a direct connection to the LDAP (where users, roles and organisations objects are stored), for instance to list all organisations.
+
+#### Entrypoints
+
+The login entrypoint is `/login` but more generally, one uses the `login` GET parameter in any querystring to force login into a given application.
+As a result, the new application may generate links like these: `/newapp/frontend/?login`, for instance if some features in the frontend are only available when authenticated.
+
+Logout entrypoint is `/logout`.
+Password recovery form is available from `/console/account/passwordRecovery`.
+Account creation form can be found at `/console/account/new`.
+
```

Comment:
I think you can also refer to the existing documentation, it does not replace this one though as it reads more technical: https://github.com/georchestra/georchestra-gateway/blob/main/docs/custom_filters.adoc#addsecheadersgatewayfilter

Reply:
I'd also mention the new json versions of the headers which are used by GN already (as far as I know).